src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialog.xaml.cs

```diff
@@ -0,0 +1,117 @@
+{
+    /// <summary>
+    /// Interaction logic for AddParameterDialog.xaml
+    /// </summary>
+    internal partial class AddParameterDialog : DialogWindow
+    {
+        public readonly AddParameterDialogViewModel ViewModel;
+        private readonly IntellisenseTextBoxViewModel _typeIntellisenseTextBoxView;
+        private readonly IntellisenseTextBoxViewModel _nameIntellisenseTextBoxView;
+        private readonly Document _document;
+
+        public string OK { get { return ServicesVSResources.OK; } }
+        public string Cancel { get { return ServicesVSResources.Cancel; } }
+
+        public string TypeNameLabel { get { return ServicesVSResources.Type_Name; } }
+
+        public string ParameterNameLabel { get { return ServicesVSResources.Parameter_Name; } }
+
+        public string CallsiteValueLabel { get { return ServicesVSResources.Callsite_Value; } }
+
+        public string AddParameterDialogTitle { get { return ServicesVSResources.Add_Parameter; } }
+
+        public AddParameterDialog(
+            IntellisenseTextBoxViewModel typeIntellisenseTextBoxViewModel,
+            IntellisenseTextBoxViewModel nameIntellisenseTextBoxViewModel,
+            INotificationService notificationService,
+            Document document)
+        {
+            // The current implementation supports Add only.
+            // The dialog should be initialized the other way if called for Edit.
+            ViewModel = new AddParameterDialogViewModel(notificationService);
+            _typeIntellisenseTextBoxView = typeIntellisenseTextBoxViewModel;
+            _nameIntellisenseTextBoxView = nameIntellisenseTextBoxViewModel;
+            _document = document;
+            this.Loaded += AddParameterDialog_Loaded;
+            DataContext = ViewModel;
+
+            InitializeComponent();
+        }
+
+        private void AddParameterDialog_Loaded(object sender, RoutedEventArgs e)
+        {
+            IntellisenseTextBox typeTextBox = new IntellisenseTextBox(
+                _typeIntellisenseTextBoxView, TypeContentControl);
+            this.TypeContentControl.Content = typeTextBox;
+
+            IntellisenseTextBox nameTextBox = new IntellisenseTextBox(
+                _nameIntellisenseTextBoxView, NameContentControl);
+            this.NameContentControl.Content = nameTextBox;
+        }
+
+        private void OK_Click(object sender, RoutedEventArgs e)
+        {
+            ViewModel.TypeName = ((IntellisenseTextBox)TypeContentControl.Content).Text;
+            ViewModel.ParameterName = ((IntellisenseTextBox)NameContentControl.Content).Text;
+
+            if (ViewModel.TrySubmit(_document))
+            {
+                DialogResult = true;
+            }
+        }
+
+        private void Cancel_Click(object sender, RoutedEventArgs e)
+        {
+            DialogResult = false;
+        }
+
+        private void TypeNameContentControl_PreviewKeyDown(object sender, KeyEventArgs e)
```

Comment:
Is this actually for just the type name or also the parameter name? Needs a rename?